doc/compiler/il/Node.md

```diff
@@ -0,0 +1,91 @@
+# Node
+
+## Description
+
+When an optimizer makes a transofrmation, a node is often transmuted from one opcode value to another. This can lead to a variety of bugs since the properties that belonged to the original node are implicitly inherited by the transmuted node. The old properties might have no meaning, or worse, a different meaning for the new transmuted node.
```

Comment:
* Please try and not use word wrap so that comments can be associated closer to the text they are about.

* Fix spelling of "transformation"